src/Shared/PackageManagers/NuGetProjectManager.cs

```diff
@@ -44,6 +45,19 @@ public class NuGetProjectManager : TypedManager<NuGetPackageVersionMetadata>
         {
             GetRegistrationEndpointAsync().Wait();
         }
+        
+        /// <inheritdoc />
+        public override async IAsyncEnumerable<NuGetBaseArtifactUri> GetArtifactDownloadUrisAsync(PackageURL purl)
+        {
+            string feedUrl = (purl.Qualifiers?["repository_url"] ?? NUGET_DEFAULT_CONTENT_ENDPOINT).EnsureTrailingSlash();
+
+            string artifactUri = $"{feedUrl}{purl.Name.ToLower()}/{purl.Version}/{purl.Name.ToLower()}.{purl.Version}.nupkg";
+            HttpClient httpClient = CreateHttpClient();
```

Comment:
Please remove this. Adding a network call and potentially downloading 100s of MBs just to throw away is huge overhead to have here for ~very little benefit I can see.

Reply:
This was in response to https://github.com/microsoft/OSSGadget/pull/329#discussion_r853207419
I could change it to a request for just the headers to see if we 404 or not. We just want to see if the uri is even valid.

Reply:
I could change it to:
` HttpResponseMessage result = await httpClient.GetAsync(artifactUri, HttpCompletionOption.ResponseHeadersRead);`
https://docs.microsoft.com/en-us/dotnet/api/System.Net.Http.HttpCompletionOption?view=net-6.0#fields

Reply:
I added a more in depth comment to that as well. I don't believe we should even be doing that.

However, if you believe it is needed, and it is possible to reduce the network calls to be of minimal size, you will need to change artifact uri and still return those Uris that are inaccessible but have a field in the artifact uri for the status code you got for the headers. 

I understand why a user would want this logic. It just doesn't belong here.

Reply:
We've used that in CFS before. This allows us to get a response to see if a given URL 404s, without waiting to retrieve the content. I'm also ok with this being up to the consumer, where they have to check if the uri being returned exists or not. 
We could also add a parameter to the method to check if the uri exists, and set a default to true or false depending on what you think.

Reply:
I'm now leaning towards removing it and leaving it up to the consumer to implement. If for whatever reason the call fails, we don't want to falsely claim the artifact doesn't exist. And I don't think it's OSS Gadget's job to add the retry logic to assert that the artifact does or doesn't exist.

Reply:
Retry logic is another good point.  This adds a lot of uncertainty to this api.

Another example, I think it is theoretically possible you could have a repository where to access packages requires a secret in the headers with different permissions than to read.

It seems to be to be preferable for the failure to come when the consumer of the api tries to get the package and can perform their own retry logic as appropriate rather than to pretend the uri doesn't exist.

Reply:
Yeah, private package registries/repositories would mess with all of this.

Reply:
I reverted it. Resolved.